src/fight/shooting_tools.rs

```diff
@@ -0,0 +1,261 @@
+///
+/// weapon_aim (isize): The aim of the weapon
+///
+/// weapon_strength (isize): The strength of the weapon
+///
+/// weapon_ap (isize): The armour penetration of the weapon
+///
+/// # Return
+/// (usize, f64): A tuple with first the damage computed and then the probability that it occurs
+fn compute_mean_case_shooting(
+    attacking_regiment: &crate::regiment::Regiment,
+    defending_regiment: &crate::regiment::Regiment,
+    weapon_aim: isize,
+    weapon_strength: isize,
+    weapon_ap: isize,
+) -> (usize, f64) {
+    //let attacking_stats: &stat::Stats = attacking_regiment.get_model().get_boosted_stats();
+    let defending_stats: &stat::Stats = defending_regiment.get_model().get_boosted_stats();
+
+    let damage_probability: f64 = compute_damage_probability_shooting(
+        defending_stats,
+        weapon_aim,
+        weapon_strength,
+        weapon_ap,
+    );
+
+    let nb_attacks: f64 = (weapon_aim as f64 * 1.5 * attacking_regiment.get_cols() as f64).round(); // TODO: 1 is default, add the number of attacks depending on the profile of the weapon attached to the unit
+    let damage: usize = std::cmp::min(
+        (nb_attacks * damage_probability).round() as usize,
+        defending_stats.get_health_points() * defending_regiment.get_nb_models(),
+    );
+
+    (
+        damage,
+        crate::math_tools::compute_bernoulli(nb_attacks as usize, damage, damage_probability),
+    )
+}
+
+/// Compute the average damage dealt by a unit to another according to the requested scenario
+///
+/// # Parameters
+/// attacking_regiment (&regiment::Regiment): The attacker
+///
+/// defending_regiment (&regiment::Regiment): The defender
+///
+/// case (&ComputeCase): The scenario from first_unit point of view
+///
+/// # Return
+/// (usize, f64): The average amount of damage dealt by first_unit and the probability for this scenario to occurs
+pub fn compute_case_shooting(
+    attacking_regiment: &crate::regiment::Regiment,
+    defending_regiment: &crate::regiment::Regiment,
+    case: &ComputeCase,
+) -> (usize, f64) {
+    // shooting stat block
+    let weapon_aim: isize;
+    let weapon_strength: isize;
+    let weapon_ap: isize;
+    //let mut weapon_shots: isize = 0;
+
+    // fetch the weapon modifiers
+    let attacking_modifiers_weapon = attacking_regiment.get_model().get_modifiers();
+    match &attacking_modifiers_weapon[0] {
+        // iterate throughout every values of Modifier::Weapon and attribute it to different variables, the values are the following
+        Modifier::Weapon(weapon_stats) => {
+            weapon_aim = weapon_stats.get_aim().unwrap_or(0);
+            //weapon_shots = weapon_stats.get_shots().unwrap_or(0);
+            weapon_strength = weapon_stats.get_strength();
+            weapon_ap = weapon_stats.get_armour_penetration();
+        }
+
+        Modifier::Global(_) => todo!(),
+        Modifier::Offensive(_) => todo!(),
+        Modifier::Defensive(_) => todo!(),
```

Comment:
why are we letting them as todo ? can those modifiers have an influence on this function ? (real question, idk)

Reply:
I used the match wrongly, I was forced to iterate to deconstruct to Modifier, but I can just select Modifier::weapon and pull out the variable that I want